### PR TITLE
Bug-14: Login-Validation

### PR DESCRIPTION
**Pull request by melonholy**

### CODE REVIEW

File: api/middlewares/errorHandler.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 const errorHandlerMiddleware = (error, req, res, next) => {
   res
     .status(error.status)
-    .json({ name: error.name, status: error.status, message: error.message });
+    .json({
+      name: error.name,
+      status: error.status,
+      message: error.message,
+      type: error.type,
```
Comment (OlegPanasyuk) [inline api/middlewares/errorHandler.js]:
```
{
 ...error,
}
```
Contract of error is look like that `{ name, status, message }`

File: api/services/errorHandlers/applicationError.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
 class ApplicationError extends Error {
-  constructor(message, status) {
+  constructor(message, status,type) {
```
Comment (OlegPanasyuk) [inline api/services/errorHandlers/applicationError.js]: Check your prettier

**Review by OlegPanasyuk — CHANGES_REQUESTED** [review]

File: api/middlewares/errorHandler.js (hunk 1 above)
Comment (melonholy) [reply api/middlewares/errorHandler.js]: Sorry that i did't inform you but we really need this field for error validation on frontend part

Comment (OlegPanasyuk) [reply api/middlewares/errorHandler.js]: You can just use a new status code to manage this error

**Review by slavasgod — APPROVED** [review]

File: api/middlewares/errorHandler.js [hunk 3]
```diff
@@ -1,7 +1,11 @@
 const errorHandlerMiddleware = (error, req, res, next) => {
-  res
-    .status(error.status)
-    .json({ name: error.name, status: error.status, message: error.message });
+  const { name, status, message, type } = error;
+  res.status(error.status).json({
+    name: name,
```
Comment (universezzz) [inline api/middlewares/errorHandler.js]: {name,status,message,type}

**Review by universezzz — APPROVED** [review]

File: api/middlewares/errorHandler.js (hunk 1 above)
Comment (melonholy) [reply api/middlewares/errorHandler.js]: Done

**Review by OlegPanasyuk — APPROVED** [review]

**Pull request merged**